website/src/examples/carto-sql-layer.js

```diff
@@ -0,0 +1,112 @@
+  'Automotive Fuel': 'aut',
+  'Eating Places': 'eap',
+  'Grocery and Food Stores': 'gro'
+};
+
+class CartoSQLDemo extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { 
+      viewState: {
+        latitude: 40.7368521,
+        longitude: -73.9936065,
+        zoom: 11,
+        pitch: 60,
+        bearing: 0
+      }
+    };
+  }
+
+  componentDidMount() {
+    this.animate();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!prevProps.params.rotate.value && this.props.params.rotate.value) {
+      this.animate();
+    }
+  }
+
+  animate() {
+    const enableAnimation = this.props.params.rotate.value;
+    if (enableAnimation) {
+      const {viewState} = this.state;
+      viewState.bearing += 0.03;
+      this.setState({ initialViewState: {...viewState }});
+      window.requestAnimationFrame(this.animate.bind(this));
+    }
+  }
+
+  static title = 'Mastercard Index In NewYork';
+  
+  static code = `${GITHUB_TREE}/examples/website/carto-sql`;
+
+  static parameters = {
+    index: {displayName: 'Index', type: 'select', options: Object.keys(INDEXES), value: 'Total Spend'},
+    industry: {displayName: 'Industry', type: 'select', options: Object.keys(INDUSTRIES), value: 'Total Retail'},
+    rotate: {displayName: 'Rotation animation', type: 'checkbox', value: true},
+  };
+
+  static mapStyle = 'https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json';
+
+  static renderInfo() {
+    return (
+      <div>
+        <p>Mastercard Index January 2020 using a z18 QuadGrid. A locations’s  index shows how it ranks against all the other locations in the area. The number represents the percentile, for example, a 900 spend index means a location has higher spend than 90% of locations.</p>
+        <div style={{height: 8, width: '100%', display: 'flex', flexDirection: 'row'}}>
+          <div style={{flex: '1 1 auto', background: 'rgb(255, 221, 154)'}}></div>
+          <div style={{flex: '1 1 auto', background: 'rgb(255, 194, 133)'}}></div>
+          <div style={{flex: '1 1 auto', background: 'rgb(250, 138, 118)'}}></div>
+          <div style={{flex: '1 1 auto', background: 'rgb(241, 109, 122)'}}></div>
+          <div style={{flex: '1 1 auto', background: 'rgb(225, 83, 131)'}}></div>
+        </div>
+        <p className="layout">
+          <span className="col-1-2">Lower index</span>
+          <span className="col-1-2" style={{textAlign: 'right'}}>Higher index</span>
+        </p>
+        <p>
+          Data source:{' '}
+          <a href="https://carto.com/solutions/mastercard/">
+            MasterCard
+          </a>
+        </p>
+      </div>
+    );
+  }
+
+  render() {
+    const {params} = this.props;
+    const {viewState} = this.state;
+    const index = INDEXES[params.index.value];
+    const industry = INDUSTRIES[params.industry.value];
+    const rotate = params.rotate.value;
+
+    return (
+      <App
+        {...this.props}
+        viewState={{...viewState}}
```

Comment:
This completely disables user control of the map (pan, zoom etc.). Is that the intention? Take a look at [the point cloud example](https://deck.gl/examples/point-cloud-layer/)'s treatment (auto rotating until user engages). [implementation](https://github.com/visgl/deck.gl/blob/master/examples/website/point-cloud/app.js#L46-L55)

Ideally animation can be handled inside the `examples` directory in case the user wants to know how it's implemented